sqlobject/src/main/java/org/jdbi/v3/sqlobject/config/internal/RegisterCollectorImpl.java

```diff
@@ -0,0 +1,35 @@
+package org.jdbi.v3.sqlobject.config.internal;
+
+import org.jdbi.v3.core.collector.JdbiCollectors;
+import org.jdbi.v3.core.config.ConfigRegistry;
+import org.jdbi.v3.core.extension.SimpleExtensionConfigurer;
+import org.jdbi.v3.sqlobject.config.RegisterCollector;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+
+public class RegisterCollectorImpl extends SimpleExtensionConfigurer {
+    @Override
+    public void configure(ConfigRegistry config, Annotation annotation, Class<?> extensionType) {
+        RegisterCollector registerCollector = (RegisterCollector) annotation;
+        JdbiCollectors collectors = config.get(JdbiCollectors.class);
+
+        try {
+            Type type = null; //resultant type
+            for(Type t : registerCollector.value().getGenericInterfaces()) {
```

Comment:
This looks like it could be something along the lines of `GenericTypes.findGenericParameter(registerCollector.value(), Collector.class).orElseThrow(...)`

Reply:
This calls `Class#getGenericSuperclass` instead of `Class#getGenericInterfaces()`, not applicable as far as I know